src/MultiFormats/Cryptography/KeccakManaged.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            base.HashCore(array, ibStart, cbSize);
+            if (cbSize == 0)
+            {
+                return;
+            }
+            
+            var sizeInBytes = SizeInBytes;
+            if (Buffer == null)
+            {
+                Buffer = new byte[sizeInBytes];
+            }
+            
+            var stride = sizeInBytes >> 3;
+            var utemps = new ulong[stride];
+            if (BuffLength == sizeInBytes)
+            {
+                throw new Exception("Unexpected error, the internal buffer is full");
+            }
+            
+            AddToBuffer(array, ref ibStart, ref cbSize);
+            if (BuffLength == sizeInBytes) //buffer full
+            {
+                System.Buffer.BlockCopy(Buffer, 0, utemps, 0, sizeInBytes);
+                KeccakF(utemps, stride);
+                BuffLength = 0;
+            }
+
+            for (; cbSize >= sizeInBytes; cbSize -= sizeInBytes, ibStart += sizeInBytes)
+            {
+                System.Buffer.BlockCopy(array, ibStart, utemps, 0, sizeInBytes);
+                KeccakF(utemps, stride);
+            }
+
+            if (cbSize <= 0)
+            {
+                return;
+            }
+            
+            System.Buffer.BlockCopy(array, ibStart, Buffer, BuffLength, cbSize);
+            BuffLength += cbSize;
+        }
+
+        protected override byte[] HashFinal()
+        {
+            var sizeInBytes = SizeInBytes;
+            var outb = new byte[HashByteLength];
+
+            //    padding
+            if (Buffer == null)
+            {
+                Buffer = new byte[sizeInBytes];
+            }
+            else
+            {
+                Array.Clear(Buffer, BuffLength, sizeInBytes - BuffLength);
+            }
+            
+            Buffer[BuffLength++] = 1;
+            Buffer[sizeInBytes - 1] |= 0x80;
+            var stride = sizeInBytes >> 3;
+            var utemps = new ulong[stride];
+            System.Buffer.BlockCopy(Buffer, 0, utemps, 0, sizeInBytes);
+            KeccakF(utemps, stride);
+            System.Buffer.BlockCopy(state, 0, outb, 0, HashByteLength);
+            return outb;
+        }
+
+        private void KeccakF(ulong[] inb, int laneCount)
+        {
+            while (--laneCount >= 0)
+            {
+                state[laneCount] ^= inb[laneCount];
+            }
+            
+            ulong Aba, Abe, Abi, Abo, Abu;
+            ulong Aga, Age, Agi, Ago, Agu;
+            ulong Aka, Ake, Aki, Ako, Aku;
+            ulong Ama, Ame, Ami, Amo, Amu;
+            ulong Asa, Ase, Asi, Aso, Asu;
+            ulong BCa, BCe, BCi, BCo, BCu;
+            ulong Da, De, Di, Do, Du;
+            ulong Eba, Ebe, Ebi, Ebo, Ebu;
+            ulong Ega, Ege, Egi, Ego, Egu;
+            ulong Eka, Eke, Eki, Eko, Eku;
+            ulong Ema, Eme, Emi, Emo, Emu;
+            ulong Esa, Ese, Esi, Eso, Esu;
+            var round = laneCount;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'round'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)